astroid/raw_building.py

```diff
@@ -559,6 +561,7 @@ def _astroid_bootstrapping() -> None:
         types.TracebackType,
     )
     for _type in builtin_types:
+        assert hasattr(_type, "__name__")
```

Comment:
What does `mypy` say here? `pyright` seems to understand this in online Github.

Reply:
`astroid/raw_building.py:565: error: "object" has no attribute "__name__"; maybe "__ne__" or "__new__"?  [attr-defined]`

Reply:
I have no idea what the type of `builtin_types` should be.

Reply:
```python
import types

builtin_types: tuple[type, ...] = (
    types.GetSetDescriptorType,
    types.GeneratorType,
    types.MemberDescriptorType,
    type(None),
    type(NotImplemented),
    types.FunctionType,
    types.MethodType,
    types.BuiltinFunctionType,
    types.ModuleType,
    types.TracebackType,
)

for i in builtin_types:
    print(i.__name__)

```

This works for `mypy`, so `tuple[type. ...]` I guess

Reply:
```
astroid/raw_building.py:534: error: Incompatible types in assignment (expression has type "Tuple[Type[GetSetDescriptorType], Type[GeneratorType[Any, Any, Any]], Type[MemberDescriptorType], object, Type[_NotImplementedType], Type[FunctionType], Type[MethodType], Type[BuiltinFunctionType], Type[Module], Type[TracebackType]]", variable has type "Tuple[type, ...]")  [assignment]
```

:scream:

Reply:
Hmm, I'll have a look at this this weekend. Perhaps my `mypy` was out of date?